tests/test_operation.py

```diff
@@ -2826,6 +2827,54 @@ class CustomOperator(qml.operation.Operator):
     assert qml.equal(new_op, CustomOperator(2.3, wires=0))
 
 
+@pytest.mark.parametrize(
+    "coeffs, obs",
+    [
+        ([0.5], [qml.X(1)]),
+        ([0.5, 0.5], [qml.Identity(1), qml.Identity(1)]),
+        ([1, 0.5], [qml.Identity(1), qml.X(2)]),
+        (
+            [0.25, -0.25, 0.25, -0.25],
+            [
+                qml.X(0) @ qml.Z(1) @ qml.Y(2),
+                qml.Y(0) @ qml.Z(1) @ qml.X(2),
+                qml.X(1) @ qml.Z(2) @ qml.Y(3),
+                qml.Y(1) @ qml.Z(2) @ qml.X(3),
+            ],
+        ),
+        (
+            [0.0625, 0.0625, -0.0625, 0.0625, -0.0625, 0.0625, -0.0625, -0.0625],
+            [
+                qml.Hadamard(0) @ qml.X(1) @ qml.X(2) @ qml.Y(3),
+                qml.X(0) @ qml.X(1) @ qml.Y(2) @ qml.X(3),
+                qml.X(0) @ qml.Y(1) @ qml.X(2) @ qml.X(3),
+                qml.X(0) @ qml.Y(1) @ qml.Y(2) @ qml.Y(3),
+                qml.Y(0) @ qml.X(1) @ qml.X(2) @ qml.X(3),
+                qml.Y(0) @ qml.X(1) @ qml.Hadamard(2) @ qml.Y(3),
+                qml.Y(0) @ qml.Y(1) @ qml.X(2) @ qml.Y(3),
+                qml.Y(0) @ qml.Y(1) @ qml.Y(2) @ qml.Hadamard(3),
+            ],
+        ),
+        (
+            [-0.5, 0.4, -0.3, 0.2],
+            [qml.Identity(1), qml.X(1) @ qml.Y(2), qml.Y(1) @ qml.X(2), qml.Z(1) @ qml.Z(2)],
+        ),
+    ],
+)
+def test_convert_to_hamiltonian(coeffs, obs):
+    """Test that arithmetic operators can be converted to Hamiltonian instances"""
+
+    opmath_instance = qml.dot(coeffs, obs)
+    converted_hamiltonian = convert_to_hamiltonian(opmath_instance)
+    assert isinstance(converted_hamiltonian, qml.Hamiltonian)
+
+    if not qml.operation.active_new_opmath():
```

Comment:
Since this is based off `master` where `__use_new_opmath` is set to `False` by default, I think this test is actually not testing anything at the moment

I would suggest doing something like the following after switching to the `enable_new_opmath` branch

```
opmath_was_active = qml.operation.active_new_opmath()
qml.operation.disable_new_opmath()
<test without the if>
if opmath_was_active:
    qml.operation.enable_new_opmath()
else:
    qml.operation.disable_new_opmath()
```

We could revisit the idea of a context manager to do this in tests systematically

Reply:
I just inserted the line above since `hamiltonian_instance = qml.Hamiltonian(coeffs, obs)` causes issues if the new opmath is enabled, but the tests are fine. Everything else works with the new opmath except for such a code line.
I'll modify the test using the following:

```
    opmath_was_active = qml.operation.active_new_opmath()
    qml.operation.disable_new_opmath()
    hamiltonian_instance = qml.Hamiltonian(coeffs, obs)
    if opmath_was_active:
        qml.operation.enable_new_opmath()
    else:
        qml.operation.disable_new_opmath()

```